File: java/src/main/java/mx/iteso/Deletion.java
package mx.iteso;

public class Deletion extends User {
    User user;

    public Deletion(User user) {
        this.user = user;
    }

    @Override
    public String getDescription() {
        return "Deletion permission to: " + user.getDescription();
    }

    /**
     * Publish permission obtain function
     * 
     * @return Permission boolean
     */
    public boolean publishPermission() {
        return user.publishPermission();
    }

    /**
     * Deletion permission allow function
     * 
     * @return Permission boolean
     */
    public boolean deletionPermission() {
        System.out.println("Deletion permission enabled");
        return true;
    }

    /**
     * Approval permission obtain function
     * 
     * @return Permission boolean
     */
    public boolean approvalPermission() {
        return user.approvalPermission();
    }

    /**
     * Comment permission obtain function
     * 
     * @return Permission boolean
     */
    public boolean commentPermission() {
        return user.commentPermission();
    }

}

File: python/src/main/python/Deletion.py

class Deletion():

    def __init__(self,incomingUser):
        self.user = incomingUser

    def getDescription(self):
        return "Deletion permission to: " + self.user.getDescription()

    def publishPermission(self):
        return self.user.publishPermission()

    def deletionPermission(self):
        print("Deletion permission enabled")
        return True

    def approvalPermission(self):
        return self.user.approvalPermission()

    def commentPermission(self):
        return self.user.commentPermission()


